src/PhpSessionPersistence.php

```diff
@@ -59,34 +63,55 @@ class PhpSessionPersistence implements InitializePersistenceIdInterface, Session
      * Those headers will be added programmatically to the response along with
      * the session set-cookie header when the session data is persisted.
      *
-     * @param bool $nonLocking use the non locking mode during initialization?
-     * @param bool $deleteCookieOnEmptySession delete cookie from browser when session becomes empty?
+     * @param array $session
      */
-    public function __construct(bool $nonLocking = false, bool $deleteCookieOnEmptySession = false)
+    public function __construct(array $session = [])
```

Comment:
This is a major BC break (API signature change)

Reply:
Possibly introduce a static constructor, i.e. `fromConfigArray` and use that in the factory, leaving the current constructor unchanged.

Reply:
Implemented static constructor - please check.